### PR TITLE
nomad: 0.12.9 -> 0.12.10, 1.0.2 -> 1.0.3, drop 0.11

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
Version bump. 1.0 has been around for a bit so it should be safe to make it default, 0.11 did not receive a backport for this CVE so we should drop it as it appears no longer supported.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by endocrimes** [comment]
Result of `nixpkgs-review pr 112233` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nomad (nomad_1_0)</li>
    <li>nomad_0_12</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6501,16 +6501,13 @@ in
...
   # Nomad never updates major go versions within a release series and is unsupported
   # on Go versions that it did not ship with. Due to historic bugs when compiled
   # with different versions we pin Go for all versions.
-  nomad_0_11 = callPackage ../applications/networking/cluster/nomad/0.11.nix {
-    buildGoPackage = buildGo114Package;
-    inherit (linuxPackages) nvidia_x11;
-    nvidiaGpuSupport = config.cudaSupport or false;
-  };
+  # Upstream partially documents used Go versions here
+  # https://github.com/hashicorp/nomad/blob/master/contributing/golang.md
```
Comment (endocrimes) [inline pkgs/top-level/all-packages.nix:6510]: Yay! - glad they added this, makes life easier in the future :sweat_smile:

**Review by endocrimes — APPROVED** [review]
:shipit:

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112233` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>nomad (nomad_1_0)</li>
<li>nomad_0_12</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112233` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>nomad (nomad_1_0)</li>
<li>nomad_0_12</li>
</ul>
</details>

**Comment by LeSuisse** [comment]
I played a bit with the build of Nomad 1.0.3, everything looks good :+1: 

Note this include the fix for CVE-2021-3283 that should be backported.

**Pull request merged**